command/server.go

```diff
@@ -871,6 +871,90 @@ func (c *ServerCommand) setupStorage(config *server.Config) (physical.Backend, e
 	return backend, nil
 }
 
+func (c *ServerCommand) InitListeners(config *server.Config, disableClustering bool, infoKeys *[]string, info *map[string]string) (int, error, []listenerutil.Listener, []*net.TCPAddr) {
+	clusterAddrs := []*net.TCPAddr{}
+
+	// Initialize the listeners
+	lns := make([]listenerutil.Listener, 0, len(config.Listeners))
+
+	c.reloadFuncsLock.Lock()
```

Comment:
this looks problematic. there are multiple returns below that don't look like they go through a `c.reloadFuncsLock.Lock()`

i think even if the program exits shortly after one of these returns, we should still be correct with our locking

Reply:
Good catch! I think this was a bug with the run command as well, up until now. I added a defer unlock to fix this.

Reply:
nice! i'll mark resolved then